### ISSUE TITLE
Do not use mixed units in timeit output

### ISSUE THREAD

**Issue by maxnoe** [ipython/ipython#14489]
Timeit reports mean and std dev in different units in some cases, e.g. the mean is in a different order of magnitude from the std.

```
In [1]: from time import sleep

In [2]: %timeit sleep(0.001)
1.08 ms ± 11.4 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

This is pretty hard to parse and violates best practices of reporting value and uncertainty in the same units.

The preferred output would be:
```
In [1]: from time import sleep

In [2]: %timeit sleep(0.001)
1.080 ± 0.011 ms per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

**Comment by Carreau** [comment]
I think this is on purpose because when there is several order of magnitude difference, same unit would becomes ever more unreadable:

```
In [5]: %timeit -r2 -n2 sleep(10)
10 s ± 0.000013 s per loop (mean ± std. dev. of 2 runs, 2 loops each)
```

I think it's probably rare to have such a difference, so if someone send a PR, I think that would be fine.

**Comment by kyracho** [comment]
Hello, looking at the codebase for the first time, it seems like the `_format_time` function is designed to be called separately for the mean and standard deviation, which means the units for each are computed independently. This leads to situations where the mean and standard deviation are displayed in different units if their magnitudes differ by a lot.

**Comment by rainyl** [comment]
I noticed that an option `-u` exists in the command line interface of python timeit, so why dont't we add the same option?
- https://docs.python.org/3/library/timeit.html#command-line-interface

**Comment by aryansk** [comment]
I took a look and have a fix locally. I’ll open a PR shortly for review.
